packages/lisk-transaction-pool/test/transaction_pool.ts

```diff
@@ -334,4 +348,112 @@ describe('transaction pool', () => {
 			expect(removeTransactionsFromQueuesStub).to.be.calledOnce;
 		});
 	});
+
+	describe('#validateReceivedTransactions', () => {
+		const validTransactions = transactions.slice(0, 2);
+		const invalidTransactions = transactions.slice(2, 5);
+		const transactionToValidate = [
+			...validTransactions,
+			...invalidTransactions,
+		];
+		let validateReceivedTransactions: () => Promise<ReadonlyArray<Transaction>>;
+		// Dummy functions to check used for assertions in tests
+		const checkForTransactionInvalidTransactionId = sandbox.stub();
+		const checkForTransactionValidTransactionId = sandbox.stub();
+
+		const validateTransactionsResponse: CheckTransactionsResult = {
+			status: false,
+			transactionsResponses: [
+				{
+					id: invalidTransactions[0].id,
+					status: false,
+					errors: [new Error(), new Error()],
+				},
+				{
+					id: invalidTransactions[1].id,
+					status: false,
+					errors: [new Error(), new Error()],
+				},
+				{
+					id: validTransactions[0].id,
+					status: true,
+					errors: [],
+				},
+				{
+					id: validTransactions[1].id,
+					status: true,
+					errors: [],
+				},
+				{
+					id: invalidTransactions[2].id,
+					status: false,
+					errors: [new Error()],
+				},
+			],
+		};
+
+		beforeEach(async () => {
+			validateTransactionsStub.returns(validateTransactionsResponse);
+			(transactionPool.queues.received.peekUntil as sinon.SinonStub).returns(
+				transactionToValidate,
+			);
+			validateReceivedTransactions = (transactionPool as any)[
+				'validateReceivedTransactions'
+			].bind(transactionPool);
+		});
+
+		it('should remove invalid transactions from the received queue', async () => {
+			checkerStubs.checkTransactionForId
+				.onFirstCall()
+				.returns(checkForTransactionInvalidTransactionId);
+			await validateReceivedTransactions();
+			expect(checkerStubs.checkTransactionForId).to.be.calledWith(
+				invalidTransactions,
+			);
+			expect(transactionPool.queues.received.removeFor).to.be.calledWith(
+				checkForTransactionInvalidTransactionId,
+			);
+		});
```

Comment:
check invalid transactions are not in the validated queue?

Reply:
Since queues are stubbed, it's only possible to check that the functions are called with correct parameters. How do you think we can check whether invalid transactions exist in the validated queue or not? I think we really need integration tests for transaction pool for testing the functionality thoroughly. :)